torch/prototypes/nestedtensor/codegen/tensor_list.py

```diff
@@ -0,0 +1,132 @@
+from string import Template
+import torch.prototypes.nestedtensor.codegen as codegen
+import argparse
+
+def get_header():
+    header = """
+#include <torch/csrc/utils/pybind.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <ATen/ATen.h>
+
+#include <vector>
+#include <iostream>
+
+namespace torch {
+namespace prototypes {
+namespace nestedtensor {"""
+    return header
+
+
+def get_footer():
+    footer = """
+// prototypes methods on torch._C
+static PyMethodDef methods[] = {
+    {"_nestedtensor_init", (PyCFunction)nestedtensor_init, METH_NOARGS, nullptr},
+    {nullptr, nullptr, 0, nullptr}};
+
+PyMethodDef* python_functions() {
+  return methods;
+}
+
+} // namespace nestedtensor
+} // namespace prototypes
+} // namespace torch
+"""
+    return footer
+
+
+def build_unary_functions():
+    cpp_template = """
+void nestedtensor_${op}(const std::vector<at::Tensor>& input1,
+                       std::vector<at::Tensor>& out) {
+  for (int64_t i = 0; i < input1.size(); i++) {
+    at::${op}_out(out[i], input1[i]);
+  }
+}"""
+    cpp_template = Template(cpp_template)
+    cpp_source = ""
+    for op in codegen.get_unary_functions():
+        cpp_source += cpp_template.substitute(op=op)
+    return cpp_source + "\n"
+
+def build_binary_functions():
+    cpp_template = """
+void nestedtensor_${op}(const std::vector<at::Tensor>& input1,
+                       const std::vector<at::Tensor>& input2,
+                       std::vector<at::Tensor>& out) {
+  for (int64_t i = 0; i < input1.size(); i++) {
+    at::${op}_out(out[i], input1[i], input2[i]);
+  }
+}"""
+    cpp_template = Template(cpp_template)
+    cpp_source = ""
+    for op in codegen.get_binary_functions():
+        cpp_source += cpp_template.substitute(op=op)
+    return cpp_source + "\n"
+
+def build_comparison_functions():
```

Comment:
this is almost identical to the `build_binary_function` version

Reply:
See above comment